src/software/thunderscope/thunderscope.py

```diff
@@ -215,7 +221,7 @@ def load_layout(self, filename=None):
                         default_shelf[key] = val
                     default_shelf.sync()
 
-    def register_refresh_function(self, refresh_func):
+    def register_refresh_function(self, refresh_func: Any) -> None:
```

Comment:
nit: Im not fully familiar with typehinting for callback functions, but I think it should be something like
```suggestion
    def register_refresh_function(self, refresh_func: Callable[[], None]) -> None:
```